### ISSUE TITLE
Links as Buttons Activation via Spacebar

### ISSUE THREAD

**Issue by techgique** [CDRH/orchid#145]
Buttons are supposed to be able to be activated by spacebar for a11y. I think we should add some JavaScript to site-wide Orchid that selects all `<a class="btn" ...>` elements and makes a spacebar press while focused behave like a click and pressing enter.

In addition, we could add some code that adds `role="button"` if it isn't already present for these elements.

**Comment by jduss4** [comment]
I think the idea of adding `role="button"` to all the places makes the most sense to me, if that will give them the behavior they are supposed to have!

**Comment by jduss4** [comment]
Okay, I've been doing a little research and it looks like role=button will make the spacebar operate similarly to the enter key for some screen readers, but not most browsers, and we will probably need to add a little javascript.

I gave it a few goes this morning but haven't managed to get it working, despite figuring out that jquery's `click()` and trigger click stuff does not actually "click" the link, but rather notifies functions waiting for an onclick type event.  I feel like there must be some combination of `.click() { this.click() }` or something that will do what we need, but for now I am going to set this aside and move on to something else!

This is a starting point for some JS that will grab anchor tag spacebar keys and prevent it from scrolling down the page.

```
$("a[role='button']").keydown(function(event) {
  if (event.keyCode == 32) {
    event.preventDefault();
  }
})
```

**Comment by jduss4** [comment]
Branch where I identified some anchor / link_to's styled as buttons:

https://github.com/CDRH/orchid/tree/a11y/role_button

**Comment by techgique** [comment]
Partially addressed in https://github.com/CDRH/orchid/pull/225

We'll want to add the JS for keyboard a11y later :+1:

**Comment by wkdewey** [comment]
Is this already implemented, or implemented by default in Chrome? I tested https://whitman-dev.unl.edu/whitmanarchive_rails/manuscripts/ (where Orchid hasn't been updated with my PR) and this functionality works.

**Comment by techgique** [comment]
> Is this already implemented, or implemented by default in Chrome? I tested https://whitman-dev.unl.edu/whitmanarchive_rails/manuscripts/ (where Orchid hasn't been updated with my PR) and this functionality works.

That is a native `<button>` element, so it makes sense it works without the PR. Looking at the PR and other work on the same branch now :+1:

**Comment by wkdewey** [comment]
Why was it an issue then? I did test with a log statement and got it to call my code.

**Comment by techgique** [comment]
The issue is with `<a>` elements. They don't activate with a spacebar press by default, and when they are styled like buttons and given the `role="button"` attribute, that's an a11y problem.

I also see that either I did a terrible job of looking for links missing the `role="button"` attribute, or additional links were added since then that didn't include it. It's probably the former though :man_facepalming:

We should probably fix those up as part of this PR as well. I think we need to look through every instance of `link_to` and ensure it has `role: "button"` in its attributes
